tests/integration/multi_page/test_pages_order.py

```diff
@@ -24,6 +23,35 @@ def test_paor001_order(dash_duo, clear_pages_state):
         order=0,
         id="multi_layout3",
     )
+    # Test more than 10 pages to ensure the supplied order is sorted correctly
+    for order_int in range(3, 13):
+        dash.register_page(
+            f"multi_layout{order_int + 1}",
+            layout=html.Div(
+                f"text for multi_layout{order_int + 1}",
+                id=f"text_multi_layout{order_int + 1}",
+            ),
+            order=order_int,
+            id=f"multi_layout{order_int + 1}",
+        )
+    # Test fractional ordering
+    dash.register_page(
+        "multi_layout3.5",
+        layout=html.Div("text for multi_layout3.5", id="text_multi_layout3.5"),
+        order=3.5,
+        id="multi_layout3.5",
+    )
+    # Test no order given
+    dash.register_page(
+        "multi_layout14",
+        layout=html.Div("text for multi_layout14", id="text_multi_layout14"),
+        id="multi_layout14",
+    )
+    dash.register_page(
+        "multi_layout15",
+        layout=html.Div("text for multi_layout15", id="text_multi_layout15"),
+        id="multi_layout15",
+    )
```

Comment:
I've just pushed a test for the first point you raised.
I'm not sure how to go about the second one. Isn't that sort of already done by `multi_layout14` and `multi_layout15`?
These don't have an `order` defined but I'm not sure I understand what you'd like to see with the three modules you mentioned... Maybe you could quickly add them yourself?

Reply:
Added a couple more, hopefully more or less self-documenting.

@AnnMarieW all of these `register_page` calls include an `id` that matches `module` but I can't tell what purpose this serves?

Reply:
interesting, so it's actually [not behaving as I thought it would](https://app.circleci.com/pipelines/github/plotly/dash/4654/workflows/e86d52a8-d1b7-44d9-9a78-ea71969f373e/jobs/78065/tests):
```
  -  'yyy_last_string_order',
     'aaa_first_with_no_order',
     'multi_layout14',
     'multi_layout15',
     'pages.defaults',
     'pages.metas',
     'pages.not_found_404',
     'pages.page1',
     'pages.page2',
     'pages.path_variables',
     'pages.query_string',
     'pages.redirect',
  +  'yyy_last_string_order',
```
ie pages with strings for `order` appear to be still interleaving with the module names. Maybe we're OK with that and can adjust the test to match, or maybe we need to fix the behavior - either way we should make this a conscious decision.

Reply:
Fixed in https://github.com/plotly/dash/pull/2565/commits/ae8ab8d16d782002496dd9014af9270beb83842f - as described in my now-hidden comment:
> OK I see what was happening, the entry in `page_registry` always has `order`, it's just `None` if it wasn't provided - so `"order" not in i` was always `False`.

@AnnMarieW to my mind this is better behavior, and can be considered 100% bugfix, would you agree?

@gothicVI the last thing we'll need in this PR is a changelog entry.

Reply:
@alexcjohnson done - also squashed all of the commits to allow for a proper merge